### ISSUE TITLE
APU sPPT on Cezanne 

### ISSUE THREAD

**Issue by bolotoff** [FlyGoat/RyzenAdj#172]
--apu-slow-limit works very well on Cezanne, though it takes some seconds to settle under load. For example, if I set 5900HX to 45W, it bursts to almost 80W, then settles at 51W. --vrmmax-current can keep those bursts under control somewhat. This is ASUS ROG Strix G15 Advantage Edition G513QY with very generous defaults producing way too much heat and noise:

CPU Family: Cezanne
SMU BIOS Interface Version: 18
Version: v0.8.2 
PM Table Version: 400005
|        Name         |   Value   |      Paramter      |
|---------------------|-----------|--------------------|
| STAPM LIMIT         |   216.000 | stapm-limit        |
| STAPM VALUE         |    16.267 |                    |
| PPT LIMIT FAST      |   216.000 | fast-limit         |
| PPT VALUE FAST      |    29.244 |                    |
| PPT LIMIT SLOW      |   190.000 | slow-limit         |
| PPT VALUE SLOW      |    17.134 |                    |
| StapmTimeConst      |     1.000 | stapm-time         |
| SlowPPTTimeConst    |     5.000 | slow-time          |
| PPT LIMIT APU       |   100.000 | apu-slow-limit     |
| PPT VALUE APU       |    10.944 |                    |
| TDC LIMIT VDD       |    74.000 | vrm-current        |
| TDC VALUE VDD       |     4.624 |                    |
| TDC LIMIT SOC       |    15.000 | vrmsoc-current     |
| TDC VALUE SOC       |     1.834 |                    |
| EDC LIMIT VDD       |   127.000 | vrmmax-current     |
| EDC VALUE VDD       |   127.000 |                    |
| EDC LIMIT SOC       |    20.000 | vrmsocmax-current  |
| EDC VALUE SOC       |     0.000 |                    |
| THM LIMIT CORE      |    95.000 | tctl-temp          |
| THM VALUE CORE      |    43.138 |                    |
| STT LIMIT APU       |    53.000 | apu-skin-temp      |
| STT VALUE APU       |    28.733 |                    |
| STT LIMIT dGPU      |    65.000 | dgpu-skin-temp     |
| STT VALUE dGPU      |    27.879 |                    |
| CCLK Boost SETPOINT |    50.000 | power-saving /     |
| CCLK BUSY VALUE     |    52.271 | max-performance    |

ASUS Armoury Crate can set --apu-slow-limit using APU sPPT slider and --slow-limit using Platform sPPT slider, some fan profiles and that's all.

I have quicktested 5900HX with LINPACK to see how well it gets at lower APU sPPT:
35W @ 76GFlops
40W @ 81GFlops
45W @ 85GFlops
50W @ 88GFlops
55W @ 91GFlops
60W @ 94GFlops
65W @ 96GFlops

70W and above still makes 96GFlops as it starts to throttle.

Let me know if there is any testing I can help with.

**Comment by ntorresalberto** [comment]
Also interested in this and can test if required, posting to add some visibility

**Comment by Falcosc** [comment]
I still don't understand, what is content of request? I only see information. Should we add some of them to the Wiki, or what do you want to tell us?

@ntorresalberto what are you interested in?
@bolotoff what do you want to add/improve?

The only point I see is `though it takes some seconds to settle` but this is expected behavior for the slow value. See our PPT documentation for comparison: https://github.com/FlyGoat/RyzenAdj/wiki/Options#ppt-limit

**Comment by ntorresalberto** [comment]
@Falcosc I can tell you my interpretation... :laughing: 
The key is here:

> For example, if I set 5900HX to 45W, it bursts to almost 80W, then settles at 51W.

1. You already adressed this: I didn't know it took so long to settle, is this the same behavior for all cezannes?
If I may, this puzzles me a bit (I apologize if this is on the wiki page):
    a) How long should it take? (or is it just dependent on thermal limit?)
    b) This behavior wouldn't make much sense on battery, limiting TDP is a bad way to manually optimize for battery?
    c) The graphs from the wiki link come from a gui tool to check power? If so, could you point me to it?
2. It settles to 51W instead of <45W, is this also expected?

**Comment by Falcosc** [comment]
> @Falcosc I can tell you my interpretation... 😆 The key is here:
> 
> > For example, if I set 5900HX to 45W, it bursts to almost 80W, then settles at 51W.
> 
>     1. You already adressed this: I didn't know it took so long to settle, is this the same behavior for all cezannes?
>        If I may, this puzzles me a bit (I apologize if this is on the wiki page):
>        a) How long should it take? (or is it just dependent on thermal limit?)
>        b) This behavior wouldn't make much sense on battery, limiting TDP is a bad way to manually optimize for battery?
>        c) The graphs from the wiki link come from a gui tool to check power? If so, could you point me to it?


Yes, same like Renoir. You can still use the Renoir Tuning Guide. It would have told all your questions https://github.com/FlyGoat/RyzenAdj/wiki/Renoir-Tuning-Guide

a) configurable, but the time value is not in seconds. I have some examples in the Renoir guide
b) depends on the combination of TDP, current and temperature limits. The Renoir Tuning guide tells a story of how difficult it is to improve a device performance because removing the limits will make it worse in some conditions
c) HWinfo, I recommended it in the Renoir Tuning guide 

> 
>     2. It settles to 51W instead of <45W, is this also expected?
That's is a very common question: https://github.com/FlyGoat/RyzenAdj/wiki/FAQ#temperature-or-power-limit-does-not-work
Why is the question common? Well because there are dozens of limits you need to keep track an any of them can be triggered before your value gets used or your value limits something which isn't used on your platform
The table output tells you which limit is currently active. If value != limit then you are looking at the wrong limit. The table of the entry post for example shows the vrmmax-current limt because 127 is the current value and the limit value.
Either you did look on the wrong value or the 45 got altered to 51 or the value didn't settle yet or there is a undiscovered value in the full table dump (no labels for this output)
